data-prepper-plugins/service-map-stateful/src/main/java/org/opensearch/dataprepper/plugins/processor/ServiceMapStatefulProcessor.java

```diff
@@ -218,12 +227,17 @@ private Collection<Record<Event>> iterateProcessorState(final MapDbProcessorStat
                     parent = previousWindow.get(child.parentSpanId);
                 }
 
-
                 final String traceGroupName = getTraceGroupName(child.traceId);
                 if (traceGroupName == null || parent == null || parent.serviceName.equals(child.serviceName)) {
                     return;
                 }
 
+                final String traceIdString = Hex.encodeHexString(parent.traceId);
```

Comment:
You can avoid this unnecessary encode by using `byte[]` in `ServiceNodeData` per my other comment.